clean-data/clean-data.R

```diff
@@ -0,0 +1,980 @@
+    developerTea <- c("^developertea", "^developer's tea", "^developer tea")
+    cleanPart1 <- normalize_text(inData = cleanPart1,
+                                 columnName = "PodcastOther",
+                                 searchTerms = developerTea,
+                                 replaceWith = "Developer Tea")
+
+
+    # Clean number of hours spent learning
+
+    ## Remove the word "hour(s)"
+    hoursIdx <- cleanPart1 %>% select(HoursLearning) %>%
+        mutate_each(funs(grepl("hours", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    hoursData <- cleanPart1 %>% filter(hoursIdx) %>%
+        mutate(HoursLearning = sub("hours.*", "", HoursLearning))
+    cleanPart1 <- cleanPart1 %>% filter(!hoursIdx) %>% bind_rows(hoursData)
+
+    ## Remove hyphen and "to" for ranges of hours
+    rangeHrIdx <- cleanPart1 %>% select(HoursLearning) %>%
+        mutate_each(funs(grepl("-|to", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    rangeHrData <- cleanPart1 %>% filter(rangeHrIdx) %>%
+        mutate(HoursLearning = average_string_range(HoursLearning))
+    cleanPart1 <- cleanPart1 %>% filter(!rangeHrIdx) %>%
+        bind_rows(rangeHrData)
+
+    ## Remove hours greater than 100 hours
+    hoursLearnNumeric <- cleanPart1 %>%
+        mutate(HoursLearning = as.integer(HoursLearning))
+    weekHours <- hoursLearnNumeric %>%
+        filter(HoursLearning > 100) %>%
+        mutate(HoursLearning = as.integer(NA)) # Change to value later
+    cleanPart1 <- hoursLearnNumeric %>%
+        filter(!(ID %in% weekHours[["ID"]])) %>%
+        bind_rows(weekHours) %>%
+        mutate(HoursLearning = as.character(HoursLearning))
+
+
+    # Clean months programming
+
+    ## Change years to months
+    yearsProgramIdx <- cleanPart1 %>% select(MonthsProgramming) %>%
+        mutate_each(funs(grepl("years", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    yearsProgramData <- cleanPart1 %>% filter(yearsProgramIdx) %>%
+        mutate(MonthsProgramming = years_to_months(MonthsProgramming))
+    cleanPart1 <- cleanPart1 %>% filter(!yearsProgramIdx) %>%
+        bind_rows(yearsProgramData)
+
+    ## Remove non-numeric characters
+    cleanPart1 <- cleanPart1 %>% sub_and_rm(colName = "MonthsProgramming",
+                              findStr = "[A-Za-z ]",
+                              replaceStr = "")
+
+    ## Average the range of months
+    avgMonthIdx <- cleanPart1 %>% select(MonthsProgramming) %>%
+        mutate_each(funs(grepl("-", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    avgMonthData <- cleanPart1 %>% filter(avgMonthIdx) %>%
+        mutate(MonthsProgramming = average_string_range(MonthsProgramming))
+    cleanPart1 <- cleanPart1 %>% filter(!avgMonthIdx) %>%
+        bind_rows(avgMonthData)
+
+
+    ## Remove outlier months of programming
+    cleanPart1 <- cleanPart1 %>%
+        mutate(MonthsProgramming = remove_outlier(MonthsProgramming, 744))
+
```

Comment:
Yeah, there are always bugs, especially when using such language as R :)
I'll continue to check first dataset tomorrow.